src/js/game/LevelMVC/LevelView.js

```diff
@@ -1933,12 +1934,11 @@ module.exports = class LevelView {
         if (this.controller.levelModel.usePlayer) {
           if (distanceBetween(this.player.position, collectiblePosition) < collectibleDistance) {
             this.playItemAcquireAnimation(this.player.position, this.player.facing, sprite, () => { }, blockType);
-          } else {
-            this.collectibleItems.push([sprite, [xOffset, yOffset], blockType, collectibleDistance]);
           }
         }
       });
     }
+    this.collectibleItems.push([sprite, [xOffset, yOffset], blockType, collectibleDistance]);
```

Comment:
Why did this get moved down here?

Reply:
@Hamms 
I spoke to Josh this morning about this. This was initiated by the fact that the unit tests weren't picking it up because we weren't mocking enough data, but I ended up moving it because before it was only getting tracked in the collectibleItems array if both:
this.controller.getIsDirectPlayerControl() && this.controller.levelModel.usePlayer

The problem was that even if those checks failed, the miniblock is still created, it's just not tracked anywhere and thus lost to the void. I moved the array tracking to the bottom of the function so that regardless of whether or not we have a player that's picking them up, we still keep track of them. 

This had a side effect of making the unit test easier, so no complaints on that front either. 👍

Reply:
Won't this have the side effect of also tracking collectibles that are collected by the player immediately upon spawning?

Reply:
I tested that and it looked like the collectibleItems array was being updated correctly. The unit test has the player at [0,0] and spawns miniblocks at [0-5, 0], and then has the player walk right 3 times. The result should be 3 mini blocks picked up, and 3 mini blocks left behind (with one of the miniblocks being picked up immediately, since player at [0,0] and a block spawned at [0,0]).

The result of the unit test is that there are 3 remaining element in the collectibleItems array, as expected.

It seems to be working?

Reply:
right, but the item at [0, 0] probably isn't getting picked up until the player moves, when it should get picked up right away.

Reply:
It does get picked up right away still, I think the risk is that it will remain in the collectibleItems array until the player moves... That's a fair point, if we consider that a concern? Though if it's cleaned up on moving, and that's only a problem in levels where the player would be manually controlling the player, perhaps it's low priority? 

What do you think?

Reply:
We definitely consider that a concern for all the 2015 levels

Reply:
Well than it shall be fixed! It was more a change to appease the unit test anyway, so any actual functional concerns definitely take precedence.